circle.yml

```diff
@@ -1,9 +1,92 @@
 version: 2
+
+.x-cockroach-2.1: &cockroach-2-1
+  image: cockroachdb/cockroach:v2.1.7
+  command: start --insecure
+
+.x-cockroach-19.1: &cockroach-19-1
+  image: cockroachdb/cockroach:v19.1.1
+  command: start --insecure
+
+.x-tox-version: &tox-version
+  TOX_VERSION: 3.12.1
+
+.x-py27: &py27
+  environment:
+    TEST_ENV: py27
+    <<: *tox-version
+
+.x-py37: &py37
+  environment:
+    TEST_ENV: py37
+    <<: *tox-version
+
+.x-test-base: &test-base
+  steps:
+    - checkout
+    - run:
+        name: Wait for database
+        command: dockerize -wait tcp://127.0.0.1:26257 -timeout 120s
+    - run:
+        name: Install testrunner
+        command: pip install --user tox==${TOX_VERSION}
+    - run:
+        name: Test
+        command: ${HOME}/.local/bin/tox -e ${TEST_ENV}
+
+.x-test-py27: &test-py27
```

Comment:
This level seems kind of useless - you still have additional references to the python version in the docker images below. Could we either move the circleci/python docker image versions into this level or move the `*py27` references down?

Reply:
You are right. This additional step is kind of useless. I will remove it and move the references down to the job definition. Moving the docker image versions up is not possible, because we need to inject the CockroachDB version at job level (for what I don't see a solution with yaml anchors).

This solution is not optimal when we want to test with more python versions, but I don't see an easy solution to this and, as you already said, it might be sufficient to just test with one version of python 2 and python 3.